src/Config/Auth.php

```diff
@@ -372,10 +372,31 @@ public function registerRedirect(): string
         return $this->getUrl($url);
     }
 
+    /**
+     * Accepts a string which can be an absolute URL or
+     * a named route or just a URI path.
+     * @param string $url
+     * @return string
+     */
     protected function getUrl(string $url): string
     {
-        return strpos($url, 'http') === 0
-            ? $url
-            : rtrim(site_url($url), '/ ');
+        // To accommodate all url patterns
+        $final_url = '';
+
+        switch (true) {
+            case strpos($url, 'http') === 0: // URL begins with 'http'. E.g. http://example.com
+                $final_url = $url;
+                break;
+
+            case route_to($url) !== false: // URL is a named-route
+                $final_url = url_to($url);
```

Comment:
Since we are trimming in the case of URI path, existing tests will fail if we do not do so here as well.

```suggestion
                $final_url = rtrim(url_to($url), '/ ');
```